pkgs/applications/networking/bee/bee-clef.nix

```
@@ -0,0 +1,71 @@
+{ version ? "release", stdenv, lib, substituteAll
+, fetchFromGitHub, go-ethereum }:
+
+let
+
+  versionSpec = {
+    release = rec {
+      name = "bee-clef";
+      version = "0.4.4";
+      rev = "refs/tags/v${version}";
+      sha256 = "18fab0nxxsglrdykhb02m6vd7pgd6z4pqgfp5nmpxjcz7s8rfnml";
+    };
+  }.${version};
+
+in
+
+stdenv.mkDerivation {
+  pname = versionSpec.name;
+  version = versionSpec.version;
+
+  src = fetchFromGitHub {
+    owner = "ethersphere";
+    repo = "bee-clef";
+    rev = versionSpec.rev;
+    sha256 = versionSpec.sha256;
+  };
+
+  buildInputs = [ go-ethereum ];
+
+  patches = [
+    (substituteAll {
+      src = ./bee-clef.patch;
+      clef = "${go-ethereum}/bin/clef";
+      inherit outputDir;
+
+      # this fails, but i also need to somehow inject paths that are referring to files in $out 
+      #??? inherit out;
+    })
+  ];
+
+  buildPhase = " "; # we don't want the Makefile to be run
+
+  # this intentionally reproduces the same directory structure as it
+  # is in the upstream repo, but only copies the files that are
+  # needed.
+  installPhase = ''
+    mkdir -p $out/lib/systemd/system
+    cp packaging/clef-service $out
+    #TODO delme substituteAll ${./ensure-clef-account} $out/ensure-clef-account
+    cp ${./ensure-clef-account} $out/ensure-clef-account
+    cp packaging/rules.js packaging/4byte.json $out/
+    chmod +x $out/clef-service $out/ensure-clef-account
+    patchShebangs $out/
+  '';
+
+  meta = with lib; {
+    # homepage = "https://gateway.ethswarm.org/bzz/docs.swarm.eth/docs/installation/bee-clef/";
+    homepage = "https://docs.ethswarm.org/docs/installation/bee-clef";
+    description = "External signer for Ethereum Swarm Bee";
+    longDescription = ''
+clef is go-ethereum's external signer.
+
+bee-clef is a package that starts up a vanilla clef instance as a systemd service, but configured in such a way that is suitable for bee (relaxed security for automated operation).
+
+This package contains the files necessary to run the bee-clef service.
```

[indented strings](https://nixos.org/manual/nix/stable/#idm140737322056256) get rid of leading whitespace, so you can do this:

```suggestion
      clef is go-ethereum's external signer.
      
      bee-clef is a package that starts up a vanilla clef instance as a systemd service, but configured in such a way that is 
      suitable for bee (relaxed security for automated operation).

      This package contains the files necessary to run the bee-clef service.
```